python-checks/src/main/java/org/sonar/python/checks/SklearnPipelineParameterAreCorrectCheck.java

```diff
@@ -0,0 +1,320 @@
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.SubscriptionContext;
+import org.sonar.plugins.python.api.symbols.ClassSymbol;
+import org.sonar.plugins.python.api.symbols.FunctionSymbol;
+import org.sonar.plugins.python.api.symbols.Symbol;
+import org.sonar.plugins.python.api.tree.CallExpression;
+import org.sonar.plugins.python.api.tree.DictionaryLiteral;
+import org.sonar.plugins.python.api.tree.Expression;
+import org.sonar.plugins.python.api.tree.ExpressionList;
+import org.sonar.plugins.python.api.tree.Name;
+import org.sonar.plugins.python.api.tree.QualifiedExpression;
+import org.sonar.plugins.python.api.tree.RegularArgument;
+import org.sonar.plugins.python.api.tree.StringLiteral;
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.python.checks.utils.Expressions;
+import org.sonar.python.tree.DictionaryLiteralImpl;
+import org.sonar.python.tree.KeyValuePairImpl;
+import org.sonar.python.tree.ListLiteralImpl;
+import org.sonar.python.tree.TreeUtils;
+import org.sonar.python.tree.TupleImpl;
+
+@Rule(key = "S6972")
+public class SklearnPipelineParameterAreCorrectCheck extends PythonSubscriptionCheck {
+
+  public static final String MESSAGE = "Provide valid parameters to the estimator.";
+  private static final Set<String> SKLEARN_SEARCH_FQNS = Set.of(
+    "sklearn.model_selection._search.GridSearchCV",
+    "sklearn.model_selection._search.HalvingGridSearchCV",
+    "sklearn.model_selection._search.RandomizedSearchCV",
+    "sklearn.model_selection._search.HalvingRandomSearchCV");
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.CALL_EXPR, SklearnPipelineParameterAreCorrectCheck::checkCallExpression);
+  }
+
+  private static void checkCallExpression(SubscriptionContext subscriptionContext) {
+    CallExpression callExpression = (CallExpression) subscriptionContext.syntaxNode();
+
+    var parsedFunctionOptional = Optional.ofNullable(callExpression.calleeSymbol())
+      .map(Symbol::fullyQualifiedName)
+      .filter(SKLEARN_SEARCH_FQNS::contains)
+      .map(callExpr -> getStepAndParametersFromDict(callExpression))
+      .flatMap(parsedParameters -> getPipelineNameAndParsedParametersFromSearchFunctions(parsedParameters, callExpression))
+      .or(() -> Optional.ofNullable(callExpression.calleeSymbol())
+        .map(Symbol::fullyQualifiedName)
+        .filter("sklearn.pipeline.Pipeline.set_params"::equals)
+        .map(callExpr -> getStepAndParametersFromArguments(callExpression))
+        .flatMap(parsedParameters -> getPipelineNameAndParsedParametersFromPipelineSetParamsFunction(parsedParameters, callExpression)));
+
+    parsedFunctionOptional.ifPresent(
+      pipelineNameAndParsedParameters -> {
+        var parsedFunction = pipelineNameAndParsedParameters.parsedParameters();
+        var pipelineName = pipelineNameAndParsedParameters.pipelineName();
```

Comment:
Here just in case you did not know. You can access Record field directly (`.parsedParameters;`).